src/Common/Wpf/Impl/Collections/RangeObservableCollection.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+
+namespace Microsoft.Common.Wpf.Collections {
+    /// <summary>
+    /// ObservableCollection that supports bulk changes
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class RangeObservableCollection<T> : ObservableCollection<T> {
+
+        public RangeObservableCollection() { }
+
+        public RangeObservableCollection(IEnumerable<T> collection) : base(collection) { }
+
+        public void AddRange(IEnumerable<T> collection) {
+            if (collection == null) {
+                throw new ArgumentNullException(nameof(collection));
+            }
+
+            foreach (var i in collection) {
+                Items.Add(i);
+            }
+
+            OnCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
+        }
+
+        public void RemoveRange(IEnumerable<T> collection) {
```

Comment:
RemoveSet or RemoveMany is probably more appropriate name since it doesn't actually remove from 'start' to 'end' and rather removes disjoint collection of items